File: guides/micronaut-scope-types/java/src/main/java/example/micronaut/infrastructuresingleton/Robot.java
package example.micronaut.infrastructuresingleton;

import io.micronaut.context.annotation.Infrastructure;
import io.micronaut.core.annotation.NonNull;
import jakarta.inject.Singleton;

import java.util.UUID;

@Singleton
@Infrastructure
public class Robot {
    @NonNull
    private final String serialNumber;

    public Robot() {
        serialNumber = UUID.randomUUID().toString();
    }

    @NonNull
    public String getSerialNumber() {
        return serialNumber;
    }
}

File: guides/micronaut-scope-types/python/src/example/micronaut/singleton/robot.py
from uuid import uuid4

from jakarta.inject import Singleton


# tag::pkg[]
# end::pkg[]
# tag::clazz[]
@Singleton  # <1>
class Robot:
    def __init__(self):
        self._serial_number = str(uuid4())

    @property
    def serial_number(self) -> str:
        return self._serial_number
# end::clazz[]
